### ISSUE TITLE
Menu position in popup mode is misaligned unless appendTo="body" is applied

### ISSUE THREAD

**Issue by jonnomk** [primefaces/primeng#13144]
### Describe the bug

Hello PrimeNG team,

This issue has been raised before, but as yet still waiting for a resolution or a 'will not fix' and perhaps an explanation, but here goes:

I am using the p-menu component extensively in our app. When using popup=true, the alignment is off by a considerable amount.  I know the work-around is to use appendTo=body, but work-arounds normally introduce a new set of problems, in the case of your menu, accessibility is now messed up because the tabindex ordering is not what the DOM expects for a good user experience.

It seems fairly trivial to implement this in a way that works in parts of the app that don't have absolute positioning by using boundingClientRectangle of the target element and positioning using those details instead of whatever method you use?

Here's a screenshot of the issue.

![image](https://github.com/primefaces/primeng/assets/56191325/074b878e-bc6b-4c91-ab3f-43dc958addff)

### Environment

Windows 11

### Angular version

15.2.1

### PrimeNG version

15.2.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.13.0

### Browser(s)

All

### Steps to reproduce the behavior

Use the menu in popup mode in a part of the app that is absolutely positioned.

### Expected behavior

The menu should align perfectly with the button or element that triggers it.

**Comment by pfdutoit** [comment]
Also running into this issue. I've made a stackblitz fork that illustrates the tabindex and positioning issue: https://stackblitz.com/edit/yy12kk-gpzvkj?file=src%2Fapp%2Fdemo%2Ftable-basic-demo.html

**Comment by jonnomk** [comment]
Just a side note. I believe appendTo=body is great when used in modals/dialogs and has its uses, but we shouldn't have to use it for every menu instance as a work-around to this problem. It'd be great to have this issue resolved and the menu work as expected.

A large number of apps use absolute positioning to position panels in the UI and unfortunately, you're menu component doesn't work as expected in these instances.

**Comment by jonnomk** [comment]
Thanks for responding :)

You have a reference to the target element here though?

![image](https://github.com/primefaces/primeng/assets/56191325/81742c08-5e22-49d9-9f75-b41e083803c1)

**Comment by cetincakiroglu** [comment]
> Thank you for the quick response @cetincakiroglu. So I presume it would not be prudent to attempt a pull request that might fix this issue?

You can create the pr, we'll review it.

**Comment by cetincakiroglu** [comment]
> Thanks for responding :)
> 
> You have a reference to the target element here though?
> 
> ![image](https://user-images.githubusercontent.com/56191325/244014480-81742c08-5e22-49d9-9f75-b41e083803c1.png)

Yes, it's because click event is independent of the PrimeNG, that's how angular works, you can reference an element to the event just like that.

On the other hand, appendTo is not a native property.

**Comment by cetincakiroglu** [comment]
Issue tracker is used for defects only as part of our commitment to quality and continuous improvement in all areas. Enhancements are collected as valuable community feedback and managed internally so moving this enhancement ticket to our internal project management backlog.

**Comment by jonnomk** [comment]
Hello @cetincakiroglu 

I honestly think this is a defect, not an enhancement - I'm not sure the problem is understood correctly.

I pass a reference to the element I want to attach the menu to here:

![image](https://github.com/primefaces/primeng/assets/56191325/48c4d01f-c9f2-41e5-8fa6-48c8b8af7801)

...but because the panel the button is in is positioned absolutely, the alignment is wrong, as shown.

![image](https://github.com/primefaces/primeng/assets/56191325/e4de67ae-73ae-4be4-8ca6-3544672f973d)

The method that is used to calculate the position of the menu is returning incorrect values. 

I even provided a screen shot of your very own documentation page with the same issue when I use devtools to apply an absolute position to the container for the menu.

![image](https://github.com/primefaces/primeng/assets/56191325/d79d03e2-9ab1-41e6-a1ce-81f23571885f)

Please don't confuse this as an accessibility problem - the problems with accessibility are a side effect of the described issue - the issue is that if anyone uses your menu component in a div/panel that is positioned absolutely, the alignment is incorrect regardless of whether I pass a reference to the container I want to attach the menu to or not.  I don't think I should have to do this either - the screen shot below should be enough to render the menu correctly, should it not?

![image](https://github.com/primefaces/primeng/assets/56191325/c30f2c4c-85be-45b1-a586-85b29c43239f)

I believe this should be re-opened as a **defect**.

**Comment by yasmany-santalla** [comment]
:host ::ng-deep p-menu > div.p-menu.p-component.p-menu-overlay {
  top: auto !important;
  left: auto !important;
}

I fixed with somethings like this.
